docs/howto/writing-migrations.txt

```diff
@@ -184,6 +184,45 @@ the respective field according to your needs.
   migration is running. Objects created after the ``AddField`` and before
   ``RunPython`` will have their original ``uuid``’s overwritten.
 
+.. _non-atomic-migrations:
+
+Non-atomic migrations
+~~~~~~~~~~~~~~~~~~~~~
+
+On databases that support DDL transactions (SQLite and PostgreSQL), migrations
+will be run inside a transaction. For some use-cases, like performing data
+migrations on large tables, you may want to exempt a migration from being run
+in a transaction by setting the ``atomic`` class attribute to ``False``::
+
+    from django.db import migrations
+
+    class Migration(migrations.Migration):
+
+        atomic = False
+
+``RunPython`` operations will still be run inside a transaction unless
```

Comment:
Good point, thanks. I'll try changing this. I'll probably have to turn the `atomic` boolean flag of `RunPython` into a tri-state flag (yes, no, undefined - with undefined being the new default). This should still be backward compatible.